### PR TITLE
Feanil/ansible xserver

### PR DESCRIPTION
**Pull request by feanil**
Rebased from on top of master.

### CODE REVIEW

File: playbooks/roles/edxapp/vars/main.yml [hunk 1]
```diff
@@ -73,7 +73,6 @@ lms_debian_pkgs:
   - gcc
   - gfortran
   - ghostscript
-  - git
```
Comment (e0d) [inline playbooks/roles/edxapp/vars/main.yml:76]: should you also remove the github-cli from here?

File: playbooks/roles/xserver/files/sandbox.sudoers [hunk 2]
```diff
@@ -0,0 +1 @@
+makeitso ALL=(sandbox) NOPASSWD:/usr/bin/python-sandbox
```
Comment (e0d) [inline playbooks/roles/xserver/files/sandbox.sudoers]: Let's kill makeitso....

File: playbooks/roles/common/tasks/main.yml [hunk 3]
```diff
@@ -77,6 +78,24 @@
...
+  - update
+
+- name: upload ssh script
+  copy: src=git_ssh.sh dest=/tmp/git_ssh.sh force=yes owner=root group=adm mode=750
+  tags:
+  - lms
+  - cms
+  - update
+
 - include: create_venv.yml
```
Comment (jarv) [inline playbooks/roles/common/tasks/main.yml:99]: Maybe not in this branch but we should remove this and use ssh key forwarding instead where needed.

File: playbooks/roles/edxapp/vars/main.yml (hunk 1 above)
Comment (feanil) [inline playbooks/roles/edxapp/vars/main.yml:76]: I pulled the git package into common but left the rest of the packages alone.  This list could probably stand to be pruned as a separate exercise.

File: playbooks/roles/xserver/templates/xserver.conf.j2 [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+start on started edxapp
+stop on stopped edxapp
+
+respawn
+respawn limit 3 30
+
+env PID=/var/tmp/xserver.pid
+env NEW_RELIC_CONFIG_FILE={{ app_base_dir }}/newrelic.ini
+env NEWRELIC={{ venv_dir }}/bin/newrelic-admin
+env WORKERS=4
```
Comment (e0d) [inline playbooks/roles/xserver/templates/xserver.conf.j2]: Should be from config or a face.

File: playbooks/roles/xserver/tasks/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  git: dest={{ xserver_grader_dir }} repo={{ xserver_grader_source }} version={{ xserver_grader_version }}
+  environment:
+    GIT_SSH: /tmp/git_ssh.sh
+  tags:
+  - deploy
+
+- name: start xserver
+  service: name=xserver state=started
+  tags:
+  - deploy
```
Comment (jarv) [inline playbooks/roles/xserver/tasks/deploy.yml:36]: What do you think belongs in here vs. handlers/main.yml?
I followed the edxapp model in xqueue where i check out the repo in deploy.yml and then notify 'invoke all deploy handlers' but I don't really have a strong preference either way.

File: playbooks/roles/xserver/templates/xserver.conf.j2 [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+env PORT={{ xserver_port }}
+env LANG=en_US.UTF-8
+env DJANGO_SETTINGS_MODULE=xserver_aws_settings
+env SERVICE_VARIANT="xserver"
```
Comment (e0d) [inline playbooks/roles/xserver/templates/xserver.conf.j2:19]: This isn't used for anything is it?

Comment (feanil) [inline playbooks/roles/xserver/templates/xserver.conf.j2:19]: I grabbed this from stage so I'm not sure.

File: playbooks/roles/xserver/templates/xserver.conf.j2 (hunk 4 above)
Comment (feanil) [inline playbooks/roles/xserver/templates/xserver.conf.j2]: Should this match the number of hyperthreaded cores on the machine?

File: playbooks/roles/xserver/templates/xserver.conf.j2 [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+start on started edxapp
+stop on stopped edxapp
+
+respawn
+respawn limit 3 30
+
+env PID=/var/tmp/xserver.pid
+env NEW_RELIC_CONFIG_FILE={{ app_base_dir }}/newrelic.ini
+env NEWRELIC={{ venv_dir }}/bin/newrelic-admin
+env WORKERS={{ ansible_processor|length }}
```
Comment (feanil) [inline playbooks/roles/xserver/templates/xserver.conf.j2:15]: Getting the number of workers from the ansible_processor fact which is a list that corresponds to the number of physical + virtual(hypearthreaded) cores.

**Comment by feanil** [comment]
Updated with changes requested.

**Pull request closed without merging**